msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -433,6 +432,50 @@ private void initializeApplication() {
         com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Create new public client application.");
     }
 
+    @Override
+    public String getWrappedFamilyRefreshToken(@NonNull final String identifier) throws MsalClientException {
+        validateNonNullArgument(identifier, "identifier");
+        validateBrokerNotInUse();
+
+        try {
+            return mTokenShareUtility.getWrappedFamilyRefreshToken(identifier);
+        } catch (Exception e) {
```

Comment:
Q: why not catch BaseException instead? getWrappedFamilyRefreshToken is thrown BaseException. Exception might be broad..

Reply:
And nit: final

Reply:
I chose not to use `BaseException` because `Future.get()` throws numerous exceptions (docs [here](https://developer.android.com/reference/java/util/concurrent/Future#get())). If I catch these exceptions and rethrow them and catch them again [here](https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/657/files#diff-490b9a6859fc9f0cff8b15e9467b39b1R443) then the developer would have to call `msalException.getCause().getCause()` in order to see the original `Exception` (the first exception is the `MsalException`, the second exception is the wrapped `BaseException`, the third is the original error) -- ideally, the developer only calls this once and always gets the originating exception. I'm open to ideas about how to tidy this up, but just changing the exception type creates this problem.

I'll update to be `final` -- but I don't have a good resolution to your first point right now

Reply:
Partially resolved in b0fa950087e7b3c4c85ee192a9778dc784951097